File: json.clj
;(ns user)


(defn gen-code [length]

  (clojure.string/join

  (clojure.core/concat

  (clojure.core/string-to-array

  (clojure.core/filterv #(re-matches #"[a-zA-Z0-9]" %) (clojure.core/conj

  (clojure.core/mapv

  (fn [x] (rand-int 62)) ;base62

  (clojure.core/interval 0 length)))

  (clojure.core/string-to-array "abcdefghijklmnopqrstuvwxyz0123456789"))))


(defn write-file [name exp data]

  (with-open [file (clojure.java.io/writer (clojure.java.io/file name "." exp))]

  (clojure.json/writer file data)))


(defn ran-code []

  (gen-code (rand-int 10) 5))


(defn file-value [name exp data]

  (write-file name exp data))


(defn -main []

  (let [username (prnt "Введите имя - ")]

  (let [password (ran-code)

  data {:username username :password password :iflogin? true}]

  (file-value "File" "json" data))))

;-main

File: json.pl
#!/usr/bin/perl
use strict;
use warnings;

my $char = join '', [map { chr($_) } 0..255];
get_password_length();

sub get_password_length {
  my $length = random(10, 20);
  print "Password: ", $char =~ tr/0-9/a-z/, "\n";
}

sub save_to_file {
  my ($name, $ext, $data) = @_;
  open(my $fh, '>', "$name.$ext "Could not open file: $!";
  binmode($fh);
  print $fh json->dump($data), "\n";
  close $fh;
}

my $username = <STDIN>;
chomp $username;

my $password = get_password_length();

my $data = {
  username => $use rname,
  password  => $password,
  iflogin?
};

if (0) {
  save_to_file("File", "json", $data);
}